### PR TITLE
Improve docs.

### PR DESCRIPTION
**Pull request by Dirbaio**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -23,17 +23,84 @@ There's a wide range of possible implementations depending on the execution envi
...
 
 - It provides functions `acquire`, `release` and `free` that libraries can directly use.
-- It provides a way for any crate to supply an implementation. This allows "target support" crates such as architecture crates (`cortex-m`, `riscv`), RTOS bindings, or HALs for multicore chips to supply the correct impl so that all the crates in the dependency tree automatically use it.
+- It provides a way for any crate to supply an implementation. This allows "target support" crates such as architecture crates (`cortex-m`, `riscv`), RTOS bindings, or HALs for multicore chips to supply the correct implementation so that all the crates in the dependency tree automatically use it.
+
+## Usage
+
+First, add a dependency on a crate providing a critical section implementation. 
+
+Such crates often require enabling a `critical-section-*` Cargo feature to get the crate
```
Comment (adamgreig) [inline README.md:38]:
```suggestion
Such crates often require enabling a `critical-section-*` Cargo feature:
```

Comment (Dirbaio) [reply README.md]: Oops, fixed :+1:

**Comment by Dirbaio** [comment]
Okay, I believe this is ready to go. @eldruin @therealprof @ryankurte could you please take a look? :D

File: README.md [hunk 2]
```diff
@@ -23,17 +23,78 @@ There's a wide range of possible implementations depending on the execution envi
...
+
+**Do not** enable any Cargo feature in `critical-section`.
 
 ## Providing an implementation
 
+Crates adding support for a particular architecture, chip or operating system should provide a critical section implementation.
+It is **strongly recommended** to gate the implementation behind a feature, so the user can still use another implementation
+if needed (having two implementations in the same binary will cause linking to fail).
+
+Add the dependency, and a a `critical-section-*` feature to your `Cargo.toml`:
```
Comment (eldruin) [inline README.md:82]:
```suggestion
Add the dependency, and a `critical-section-*` feature to your `Cargo.toml`:
```

**Review by eldruin — COMMENTED** [review]
Looks good to me, just saw a typo, thanks!

**Review by eldruin — APPROVED** [review]
Thanks!
bors r+

**Pull request merged**